app.js

```diff
@@ -286,6 +293,43 @@ function create_db() {
 	}).end();
 }
 
+
+function get_lastpolled(repo) {
+    var deferred = new Promise(function(resolve, reject) {
+        var doc = {};
+        var opts = clone(optionsdb);
+        opts.method = 'GET';
+        opts.path = '/' + config.db.name + '/' + repo.replace(/\//g, '---');
+        var result = '2016-01-01T00:00:00Z'; // default to earliest UTC value
+
+        var req = http.request(opts, function(res){
+            res.on('data', function(chunk) {
+                if ((res.statusCode === 200) || (res.statusCode === 304)) {
+                    doc = JSON.parse(chunk);
+                    //console.log('--- GET_LASTPOLLED: old date: ',doc.repofullname, doc.date);
+                    result = doc.date;
+
+                } else {
+                    // create the lastpolled document
+                    //console.log('--- GET_LASTPOLLED: creating: ',repo);
+                    doc.type = 'lastpolled';
+                    doc.repofullname = repo;
+                }
+
+                // update / create lastpolled doc
+                opts.method = 'PUT'
+                doc.date = new Date().toISOString();
+                var db = http.request(opts, handle_response);
+                db.write(JSON.stringify(doc))
+                db.end();
+                //console.log('--- GET_LASTPOLLED: new date: ',doc.repofullname, doc.date);
+                resolve(result);
+            })
+        }).end();
+    });
+    return deferred; // <--- happens IMMEDIATELY (object that promise listens on)
+}
+
```

Comment:
This function allows load_repos to query for new commits/pull_requests since the last sync, instead of having to go through the entire history every time.